integration_oidc_test.go

```diff
@@ -264,6 +264,7 @@ oidc:
 	for hostname, tailscale := range s.tailscales {
 		s.joinWaitGroup.Add(1)
 		go s.AuthenticateOIDC(headscaleEndpoint, hostname, tailscale)
+		time.Sleep(1 * time.Second)
```

Comment:
This looks like it could be quite racy, dependent on the underlying machine etc

Reply:
This is a workaround. I tracked the issue in https://github.com/juanfont/headscale/issues/814.

It needs proper fixing there.

Reply:
Add a TODO comment and add the issue number.